.github/actions/setup-nix/action.yaml

```diff
@@ -23,7 +23,6 @@ runs:
         key: cache-nix-${{ runner.os }}-id-${{ inputs.cache-id }}-${{ hashFiles('nix/**/*.nix') }}
         restore-keys: |
           cache-nix-${{ runner.os }}-common-
-        restore-key-hit: true
```

Comment:
Why is this removed? If active a cache won't be created if it hits the key, does that cause a problem?

Reply:
> it's actually in the `‎make-similar-caches‎`, which is equivalent to our `Setup Nix` I think?

Exactly. My goal was to call the cache job as early as possible so it'll unpack cached store and prevent redownloading store entries, but also after installing nix (it's a prerequisite).

> I tried the PR and gives me the same error:[PostgREST/postgrest/actions/runs/7035767110/job/19147207413#step:12:14](https://github.com/PostgREST/postgrest/actions/runs/7035767110/job/19147207413#step:12:14). It seems that this won't/shouldn't be executed in PRs but when merging on main only.

I think the reason it fails is the `common` cache reference/ID is reserved somewhere (very) early in the run and isn't let go afterward. The fix could be to extract the job to a separate workflow and have it triggered by the, um, schedule, for instance (there are other triggers that look promising, such as a [check_run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run)).

Reply:
Hmm... I see, I'm kinda lost with how the merging is done or even if it's needed in this case, mostly because only one version of linux `ubuntu-latest` is used (correct me if I'm wrong here, not quite familiarized with that process). 

> I think the reason it fails is the common cache reference/ID is reserved somewhere (very) early in the run and isn't let go afterward.

Yeah, I don't think the `common` cache should be removed (at least in PRs), it is used later in the release too (Release-Docker).

Reply:
Ah I assumed it was more relevant for versions due to them having similar caches, it looks like it's clever in finding similar stores alongside different Job steps then. I'll try to update my PR later and see how i works.